### DISCUSSION TITLE
Nominatim tables exploitation

### DISCUSSION THREAD

**Discussion by tportal** [osm-search/Nominatim#2810] [Q&A]
Hello,
I'm looking for the list of a country addresses. So I import, using nominatim, a OSM pbf file.
Using psql, I can list the tables created by this tool but I don't find any documentation explaning how to use these data.
Can someone help me ?
Thanks
Thierry

**Comment by mtmail** [comment]
Database layout is outlined at https://nominatim.org/release-docs/latest/develop/Database-Layout/ What kind of output do you plan to extract, can you give an example?

**Reply by tportal** [reply]
I've import the PBF file with perimeter limitation of street :
**.env :** 
NOMINATIM_IMPORT_STYLE=street
**Command :**
nominatim import --osm-file centro-latest.osm.pbf --no-updates 2>&1 | tee setup.log
I'm looking for all addresses (streets names) of Italy per town.

The generated database contains the following tables :

- country_name
- country_osm_grid
- import_status
- location_postcode
- location_property_osmline
- location_property_tiger
- new_query_log
- nominatim_properties
- place_addressline
- placex
- search_name
- spatial_ref_sys
- word

And I don't know which tables contain town and theirs streets

**Reply by mtmail** [reply]
`placex` contains town and street names. For example `select ... from placex where osm_type='relation' and osm_id=42602` for Florence (https://www.openstreetmap.org/relation/42602)

>  I'm looking for all addresses (streets names) of Italy per town.

Have a look at the `nominatim export` command. You might have to change the script (https://github.com/osm-search/Nominatim/blob/master/lib-php/admin/export.php) a little but it does most of the lookups without having to run SQL queries.

An alternative might be to impor the data into photon, then export from the newly created ElasticSearch database (see https://github.com/osm-search/Nominatim/discussions/2692)

**Reply by tportal** [reply]
In the table placex for osm_type='R' and osm_id=42602, I only have 1 record which is the town Florence.

++ Do you know where I can find the town and theirs osm_id ? in Placex ?

Thanks for your help

**Reply by mtmail** [reply]
A town is usually an `admin_level=8` (in `placex` table) https://wiki.openstreetmap.org/wiki/Tag:boundary but every country has exceptions. For example some big towns might be a city and a state (not sure if that's the case in Italy) and an island might have the administrative power as a city in real life but in OSM data it's "just" an island.

**Reply by tportal** [reply]
I try a new import with perimeter addresses :-)
